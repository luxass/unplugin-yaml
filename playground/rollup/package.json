{
  "name": "rollup-playground",
  "private": true,
  "scripts": {
    "build": "webpack --mode=development"
  },
  "devDependencies": {
    "unplugin-yaml": "workspace:*",
    "webpack": "^5.38.1",
    "webpack-cli": "^4.7.2"
  }
}
