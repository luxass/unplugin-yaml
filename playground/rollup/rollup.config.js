// @ts-check

// @ts-expect-error - types is there, but this is not picking them up.
import YAMLPlugin from "../../dist/rollup.mjs";

/** @type {import("rollup").RollupOptions} */
export default {
  input: "src/index.js",
  output: {
    format: "esm",
    sourcemap: "hidden",
    dir: "dist",
  },
  plugins: [
    YAMLPlugin(),
  ],
};
