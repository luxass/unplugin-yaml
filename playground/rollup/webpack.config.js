const path = require("node:path");
const YAMLPlugin = require("../../dist/webpack.cjs");

module.exports = {
  entry: "./src/index.js",
  output: {
    filename: "index.js",
    path: path.resolve(__dirname, "dist"),
  },
  plugins: [
    YAMLPlugin(),
  ],
  module: {
    rules: [
      {
        test: /\.yaml$/,
        exclude: /node_modules/,
        use: {
          loader: path.resolve("./yaml-loader"),
        },
      },
    ],
  },
  stats: "errors-only",
};
