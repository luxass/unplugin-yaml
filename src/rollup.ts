/**
 * This entry file is for Rollup plugin.
 *
 * @module
 */

import { unplugin } from "./";

/**
 * Rollup plugin
 *
 * @example
 * ```ts
 * // rollup.config.js
 * import yaml from "unplugin-yaml/rollup"
 *
 * export default {
 *   plugins: [yaml()],
 * }
 * ```
 */
export default unplugin.rollup;
