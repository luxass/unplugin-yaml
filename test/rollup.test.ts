import { rollup } from "rollup";
import { expect, it } from "vitest";

import YAMLPlugin from "../src/rollup";
import { removeComments } from "./utils";

it("expect yaml import to be a json object", async () => {
  const bundle = await rollup({
    input: "./test/fixtures/yaml-config.yaml",
    plugins: [
      YAMLPlugin(),
    ],
  });
  const { output } = await bundle.generate({
    format: "esm",
    sourcemap: false,
  });

  expect(removeComments(output[0].code)).toMatchSnapshot();
});

it("expect yaml import to be a string", async () => {
  const bundle = await rollup({
    input: "./test/fixtures/js-yaml-raw.js",
    plugins: [
      YAMLPlugin(),
    ],
  });
  const { output } = await bundle.generate({
    format: "esm",
    sourcemap: false,
  });

  expect(removeComments(output[0].code)).toMatchSnapshot();
});

it("expect yml import to be a json object", async () => {
  const bundle = await rollup({
    input: "./test/fixtures/yml-config.yml",
    plugins: [
      YAMLPlugin(),
    ],
  });
  const { output } = await bundle.generate({
    format: "esm",
    sourcemap: false,
  });

  expect(removeComments(output[0].code)).toMatchSnapshot();
});

it("expect yml import to be a string", async () => {
  const bundle = await rollup({
    input: "./test/fixtures/js-yml-raw.js",
    plugins: [
      YAMLPlugin(),
    ],
  });
  const { output } = await bundle.generate({
    format: "esm",
    sourcemap: false,
  });

  expect(removeComments(output[0].code)).toMatchSnapshot();
});
