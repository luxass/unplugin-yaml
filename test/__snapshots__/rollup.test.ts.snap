// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expect yaml import to be a json object 1`] = `
"var data = {
  "pluginDir": "./plugins",
  "web": {
    "enabled": true
  },
  "logging": {
    "type": "stdout",
    "level": "info"
  }
};

console.log(data);
"
`;

exports[`expect yaml import to be a string 1`] = `
"var config = "pluginDir: ./plugins\\n\\nweb:\\n  enabled: true\\n\\nlogging:\\n  type: stdout\\n  level: info\\n";

console.log(config);
"
`;

exports[`expect yml import to be a json object 1`] = `
"var data = {
  "pluginDir": "./plugins",
  "web": {
    "enabled": true
  },
  "logging": {
    "type": "stdout",
    "level": "info"
  }
};

console.log(data);
"
`;

exports[`expect yml import to be a string 1`] = `
"var config = "pluginDir: ./plugins\\n\\nweb:\\n  enabled: true\\n\\nlogging:\\n  type: stdout\\n  level: info\\n";

console.log(config);
"
`;
